src/Tools/dotnet-monitor/Exceptions/ExceptionsOperation.cs

```diff
@@ -123,47 +133,133 @@ await using (Utf8JsonWriter writer = new(stream, new JsonWriterOptions() { Inden
+            Stream stream,
+            IExceptionInstance currentInstance,
+            IDictionary<ulong, IExceptionInstance> priorInstances,
+            CancellationToken token)
         {
             // This format is similar of that which is written to the console when an unhandled exception occurs. Each
             // exception will appear as:
 
-            // First chance exception. <TypeName>: <Message>
-            //   at Class.Method
+            // First chance exception at <TimeStamp>
+            // <TypeName>: <Message>
+            //  ---> <InnerExceptionTypeName>: <InnerExceptionMessage>
+            //   at <StackFrameClass>.<StackFrameMethod>(<ParameterType1>, <ParameterType2>, ...)
+            //   --- End of inner exception stack trace ---
+            //   at <StackFrameClass>.<StackFrameMethod>(<ParameterType1>, <ParameterType2>, ...)
 
             await using StreamWriter writer = new(stream, leaveOpen: true);
 
-            await writer.WriteLineAsync(
-                string.Format(
-                    CultureInfo.InvariantCulture,
-                    Strings.OutputFormatString_FirstChanceException,
-                    instance.TypeName,
-                    instance.Message));
+            await writer.WriteAsync("First chance exception at ");
+            await writer.WriteAsync(currentInstance.Timestamp.ToString("O", CultureInfo.InvariantCulture));
+
+            await writer.WriteLineAsync();
+            await WriteTextExceptionFormat(writer, currentInstance);
+
+            await WriteTextInnerExceptionsAndStackFrames(writer, currentInstance, priorInstances);
+
+            await writer.WriteLineAsync();
+            await writer.WriteLineAsync();
+
+            await writer.FlushAsync();
+        }
+
+        // Writes the inner exceptions and stack frames of the current exception:
+        // - The primary inner exception is written with a separator message.
+        // - The call stack frames are written for the current exception
+        // - The remaining inner exceptions are written with their inner exception index
+        // The above fits the format of inner exception and call stack information reported
+        // by AggregateException instances.
+        private static async Task WriteTextInnerExceptionsAndStackFrames(TextWriter writer, IExceptionInstance currentInstance, IDictionary<ulong, IExceptionInstance> priorInstances)
+        {
+            if (currentInstance.InnerExceptionIds?.Length > 0)
+            {
+                if (priorInstances.TryGetValue(
+                        currentInstance.InnerExceptionIds[0],
+                        out IExceptionInstance primaryInnerInstance))
+                {
+                    await WriteTextInnerException(writer, primaryInnerInstance, 0, priorInstances);
+
+                    await writer.WriteLineAsync();
+                    await writer.WriteAsync("   --- End of inner exception stack trace ---");
```

Comment:
nit: Did you want a small wrapper around writes to add indent levels?

Reply:
There's only one indent level and its not recursive (the indent level doesn't increase as you go deeper into the exception tree).